### PR TITLE
SALTO-7089 wait for workspaceState to finish building in workspace.cl…

### PR DESCRIPTION
**Pull request by dantal4**
…ose()

When we run a workspace method that rebuilds the workspaceState (e.g updateNaclFiles), we need to wait for the workspaceState to finish rebuilding before calling remoteMapCreator.close() , else the remote map calls in the building process will fail on closed connection.

---

_Additional context for reviewer_

---
_Release Notes_: 
Workspace:
- prevent a future bug of closing a workspace while a workspaceState is being built, causing to remoteMap calls to fail on closed connections.

---
_User Notifications_: 
None

### CODE REVIEW

**Comment by dantal4** [comment]
@ori-moisis  do you have a suggestion how to test it?

File: packages/workspace/src/workspace/workspace.ts [hunk 1]
```diff
@@ -1211,6 +1211,10 @@ export const loadWorkspace = async (
     elements: elementsImpl,
     state,
     close: async () => {
+      if (workspaceState !== undefined) {
+        // in case that `workspaceState` is currently beind built, we'd like to wait until it finishes.
+        await workspaceState
```
Comment (omrilit) [inline packages/workspace/src/workspace/workspace.ts:1216]: Maybe worth adding a log so we could understand why it may take some time to close it?

**Comment by ori-moisis** [comment]
you can spy on something that is used when building the workspace state and then use `makeResolvablePromise` or `stepManager` from `@salto-io/test-utils` to manage when each step happens and ensure the close is called before the workspace state is built

**Review by omrilit — APPROVED** [review]
🔥

**Pull request merged**